src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -147,8 +147,9 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
       resources <- if(isConstrained) accessPolicyDAO.listResourceWithAuthdomains(resourceTypeName, rids) else IO.pure(Set.empty)
       authDomainMap = resources.map(x => x.resourceId -> x.authDomain).toMap
 
-      allAuthDomainResourcesUserIsMemberOf <- if(isConstrained) accessPolicyDAO.listAccessPolicies(ManagedGroupService.managedGroupTypeName, userId) else IO.pure(Set.empty[ResourceIdAndPolicyName])
+      allAuthDomain <- if(isConstrained) accessPolicyDAO.listAccessPolicies(ManagedGroupService.managedGroupTypeName, userId) else IO.pure(Set.empty[ResourceIdAndPolicyName])
 
+      allAuthDomainResourcesUserIsMemberOf = allAuthDomain.filter(x => ManagedGroupService.userMemberShipPolicyNames.contains(x.accessPolicyName))
```

Comment:
I think this should call the managed group service to list the groups a user is in instead of duplicating the logic here. If `ManagedGroupService.listGroups` is not quite right as it is, refactor it a little. But I do see the concern that `ManagedGroupService` is already dependent on `ResourceService` and we can't have it going the other way too. Perhaps we should have a `PolicyEvaluatorService` that has `listUserResourceActions` and `hasPermission`.